src/app/core/component/cardrenderer/card-renderer.component.ts

```diff
@@ -4,33 +4,105 @@ import {ActivatedRoute, Router} from '@angular/router';
-                private router: Router,
-                private activatedRoute: ActivatedRoute) {
-    }
+  constructor(private zone: NgZone,
+              private router: Router,
+              private activatedRoute: ActivatedRoute,
+              private patternService: PatternService) {
+  }
 
-    navigate(pattern: UriEntity): void {
-        this.zone.run(() => {
-            this.router.navigate([UriConverter.doubleEncodeUri(pattern.id)], {relativeTo: this.activatedRoute});
-        });
-    }
+  navigate(pattern: UriEntity): void {
+    this.zone.run(() => {
+      this.router.navigate([UriConverter.doubleEncodeUri(pattern.id)], {relativeTo: this.activatedRoute});
+    });
+  }
 
-    getLinkCount(directedEdges: HalLink[] | HalLink): number {
-        if (!directedEdges) {
-            return 0;
+  delete(pattern: Pattern): void {
+    this.patternService.deletePattern(pattern._links.self.href)
+      .subscribe(
+        value => this.handlePatternDelete(pattern),
+        error => {
+          console.log(`Delete unsuccessful! ${error}`);
         }
-        return Array.isArray(directedEdges) ? directedEdges.length : 1;
+      );
+  }
+
+  getLinkCount(directedEdges: HalLink[] | HalLink): number {
+    if (!directedEdges) {
+      return 0;
+    }
+    return Array.isArray(directedEdges) ? directedEdges.length : 1;
+  }
+
+  /**
+   * This method is called when the pattern is successfully deleted by the pattern service.
+   * It removes the selected pattern from the uriEntities, and
+   * removes all edges from all other patterns where the pattern to delete
+   * is part of the edge.
+   * This method is only used to update the rendered information in the frontend.
+   *
+   * @param pattern: The pattern which is deleted.
+   */
+  private handlePatternDelete(pattern: Pattern): void {
+    this.uriEntities = this.uriEntities.filter(value => value !== pattern);
+    let allEdgesToRemove: HalLink[] = [];
+    const allEdgesToRemoveHref: String[] = [];
+    if (pattern._links.outgoingDirectedEdges) {
+      Array.isArray(pattern._links.outgoingDirectedEdges) ? allEdgesToRemove = [...allEdgesToRemove, ...pattern._links.outgoingDirectedEdges] :
+        allEdgesToRemove.push(pattern._links.outgoingDirectedEdges);
     }
+    if (pattern._links.ingoingDirectedEdges) {
+      Array.isArray(pattern._links.ingoingDirectedEdges) ? allEdgesToRemove = [...allEdgesToRemove, ...pattern._links.ingoingDirectedEdges] :
+        allEdgesToRemove.push(pattern._links.ingoingDirectedEdges);
+    }
+    if (pattern._links.undirectedEdges) {
+      Array.isArray(pattern._links.undirectedEdges) ? allEdgesToRemove = [...allEdgesToRemove, ...pattern._links.undirectedEdges] :
+        allEdgesToRemove.push(pattern._links.undirectedEdges);
+    }
+    allEdgesToRemove.forEach(link => allEdgesToRemoveHref.push(link.href));
+    this.uriEntities.forEach(otherPattern => {
+      if (otherPattern._links.outgoingDirectedEdges) {
```

Comment:
This method is really really long and therefore hard to read. Can we make it smaller? 
- I'd start with calling a method for each "otherPattern" in the foreach loop and
- i'd further check if it's possible to abstract some logic of the following if-blocks (essentially you do the same for undirected, outgoing/ingoing directed links)